src/emitters/sky.cpp

```diff
@@ -0,0 +1,203 @@
+#include <mitsuba/core/bsphere.h>
+#include <mitsuba/core/fresolver.h>
+#include <mitsuba/core/plugin.h>
+#include <mitsuba/core/properties.h>
+#include <mitsuba/core/warp.h>
+#include <mitsuba/render/emitter.h>
+#include <mitsuba/render/scene.h>
+#include <mitsuba/render/texture.h>
+#include "sunsky/sunmodel.h"
+#include "sunsky/skymodel.h"
+
+NAMESPACE_BEGIN(mitsuba)
+
+/**!
+
+.. _emitter-sky:
+
+Skylight emitter (:monosp:`sky`)
+-------------------------------------------------
+
+.. pluginparameters::
+
+ * - radiance
+   - |spectrum|
+   - Specifies the emitted radiance in units of power per unit area per unit steradian.
+     (Default: :ref:`emitter-d65`)
+
+This plugin implements a skylight emitter, which surrounds
+the scene and radiates diffuse illumination towards it. This is often
+a good default light source when the goal is to visualize some loaded
+geometry that uses basic (e.g. diffuse) materials.
+
+ */
+
+template <typename Float, typename Spectrum>
+class SkyEmitter final : public Emitter<Float, Spectrum> {
+public:
+    MTS_IMPORT_BASE(Emitter, m_flags)
+    MTS_IMPORT_TYPES(Scene, Shape, Texture)
+
+    SkyEmitter(const Properties &props) : Base(props) {
+        /* Until `set_scene` is called, we have no information
+           about the scene and default to the unit bounding sphere. */
+        m_bsphere = ScalarBoundingSphere3f(ScalarPoint3f(0.f), 1.f);
+
+        m_flags = +EmitterFlags::Infinite;
+
+        m_scale = props.float_("scale", 1.0f);
+        m_turbidity = props.float_("turbidity", 3.0f);
+        m_stretch = props.float_("stretch", 1.0f);
+        m_resolution = props.int_("resolution", 512);
+        m_albedo = props.texture<Texture>("albedo", 0.2f);
```

Comment:
I'm surprised this works: `m_albedo` has a `Spectrum` type, but here the properties you query has type `ref<Texture>`.
If you try to print the value of `albedo` here, are you getting what you expect?

I think the correct usage would be to change the type of `m_albedo` to `ref<Texture>`, and then query it like so in your `eval` method:

https://github.com/mitsuba-renderer/mitsuba2/blob/f30bd5cfe69340087597a9d684ec7b2c61718c4f/src/bsdfs/diffuse.cpp#L99